### PR TITLE
objurl: unify s3 and local file url representation

### PR DESCRIPTION
**Pull request by igungor**

### CODE REVIEW

File: objurl/objurl.go [hunk 1]
```diff
@@ -21,26 +22,116 @@ const (
...
+// remote storage.
+type ObjectURL struct {
+	Type      objurlType
+	Scheme    string
+	Bucket    string
+	Path      string
+	Delimiter string
+	Prefix    string
+
+	trimmedPath string
```
Comment (sonmezonur) [inline objurl/objurl.go]: I think `relativePath` would be better.

Comment (igungor) [reply objurl/objurl.go]: done.

File: objurl/objurl.go [hunk 2]
```diff
@@ -201,9 +201,9 @@ func (o *ObjectURL) Clone() *ObjectURL {
 		Path:      o.Path,
 		Prefix:    o.Prefix,
 		// FIXME(ig): TBD
```
Comment (sonmezonur) [inline objurl/objurl.go]: can we remove this line now?

Comment (igungor) [reply objurl/objurl.go]: done.

**Pull request merged**